R/utils_memoised.R

```diff
@@ -68,18 +64,28 @@ memoise_bioc_mirrors <- memoise::memoise({
 
 
 
+#' @importFrom memoise memoise
 memoise_installed_packages <- memoise::memoise({
   function(...) utils::installed.packages(...)
 })
 
 
 
+#' @importFrom memoise memoise
 memoise_available_packages <- memoise::memoise({
-  function(..., repos = getOption("repos")) {
-    if (is.null(repos))
+  function(..., repos = getOption("repos"), .local = getOption("riskmetric.tests")) {
+    if (!is.null(.local)) {
+      db <- read.csv(
+        file.path(.local, "test_webmocks", "data", "cran_packages.csv"),
+        stringsAsFactors = FALSE)
+      db[, "Repository"] <- contrib.url(repos, getOption("pkgType"))
+      return(db)
+    } else if (is.null(repos)) {
       return(utils::available.packages(NULL))
-    else if ("@CRAN@" %in% repos)
-      repos[repos == "@CRAN@"] <- "https://cran.rstudio.com/"
+    } else if ("@CRAN@" %in% repos) {
+      repos[repos == "@CRAN@"] <- "https://cloud.r-project.org"
```

Comment:
what's the rational to change `cran.rstudio.com` and `cloud.r-project.org`?

Shall we add `repo` as an argument to allow user determine the source? For example, people might be interest in the available package from an RSPM snapshot.  https://packagemanager.rstudio.com/client/#/repos/1/overview

Reply:
I just prefer to use the r-project url over a url hosted by a private organization. `repos` is an argument, 

https://github.com/pharmaR/riskmetric/blob/84d8810f9ad661db39883578b9cec82c1b3ad883/R/utils_memoised.R#L76

and defaults to `getOption("repos")` - this is the same option that a user would use to redirect to a different package manager repo. `@CRAN@` is a value that I've seen handled specifically in other packages, but ideally this option is almost always set to a url.